Add a tutorial to Pangenome analysis with Prokka & Roary (#6998)
galaxyproject/training-material: topics/genome-annotation/tutorials/pangenome-annotation-with-roary/tutorial.md

---
layout: tutorial_hands_on

title: "Pangenome analysis with Roary"
zenodo_link: "https://doi.org/10.5281/zenodo.21339040"
tags:
  - pangenome
  - prokaryote
  - phylogeny
questions:
  - "How can we identify core and accessory genes across multiple bacterial strains?"
  - "How do we generate a pangenome using Prokka and Roary?"
  - "How can we infer the evolutionary relationship of these strains?"
objectives:
  - "Annotate genomes using Prokka."
  - "Use Roary to identify core and accessory genes."
  - "Construct a phylogenetic tree of core genes with IQ-TREE."
time_estimation: "1h"
level: Introductory
key_points:
  - "Prokka is a useful tool to annotate a bacterial genome."
  - "Roary compares annotated genomes to find core and accessory genes."
  - "Core genes can be used to build a phylogenetic tree with IQ-TREE to show relationships."
contributions:
  authorship:
    - Maed0x
    - SaimMomin12
edam_ontology:
- topic_0362 # Genome annotation
- topic_0622 # Genomics

---

# Pangenomics

> When the sequencing era truly began in the 1970s with the establishment of Sanger sequencing, it laid the foundation for the Human Genome Project, which in turn led to the human reference genome as we know it today.

Reviewer: ```suggestion
When the sequencing era truly began in the 1970s with the establishment of Sanger sequencing, it laid the foundation for the Human Genome Project, which in turn led to the human reference genome used widely today.
```

<figure style="text-align: center;">
  <img src="./images/pangenome_comparison.png" alt="Comparison of a pangenome approach with traditional reference approach" style="width:100%;">
  <figcaption>
    <strong>Figure 1: Comparison of a pangenome approach with a traditional reference approach</strong> ({% cite Matthews2024 %}). With a traditional reference approach, any sample reads that cannot be mapped to the reference are discarded. A pangenomic approach solves this limitation by capturing a wider range of genetic diversity. This allows a higher percentage of sample reads to map, meaning more of the actual sample genome can be used for downstream analysis.
  </figcaption>
</figure>

> The concept of a pangenome is not brand new. However, when the term "Pangenome" was born, it was simply not financially or computationally feasible to pursue it further. "Pangenome" was first used by {% cite Sigaux2000 %} to describe a database of genomic and transcriptomic alterations found in tumor cells versus normal cells. {% cite Tettelin2005 %} also brought the concept of pangenomics into the spotlight while sequencing multiple strains of Group B Streptococcus (GBS) to help develop a vaccine. During this process, they noticed that with every new strain they sequenced, they kept finding new genes that did not exist in any of the previously sequenced strains. They realized that a single reference genome could never capture the entire genetic diversity of a species. If they had used the first sequenced strain as their sole reference, all the novel genes from the new strains would have been unmapped and discarded during analysis.

Reviewer: ```suggestion
The concept of a pangenome is not novel. However, when the term "Pangenome" was born, it was  not financially or computationally feasible to pursue it further. "Pangenome" was first used by {% cite Sigaux2000 %} to describe a database of genomic and transcriptomic alterations found in tumor cells versus normal cells. {% cite Tettelin2005 %} also brought the concept of pangenomics into the spotlight while sequencing multiple strains of Group B Streptococcus (GBS) to help develop a vaccine. During this process, they noticed that with every new strain they sequenced, they kept contined to identify novel genes absent from previously sequenced strains. They concluded that a single reference genome was insufficient to capture the full genetic diversity of a species. If they had used the first sequenced strain as their sole reference, all the novel genes from the new strains would have been unmapped and discarded during analysis.
```

> Thanks to the rise of Next-Generation Sequencing technologies around 2010, sequencing became significantly faster and more affordable. Today, the reduced sequencing costs, combined with increased computational resources compared to the early 2000s, make it possible to properly realize true pangenomes. This technological leap led to the release of the first human pangenome draft in 2023 ({% cite Liao2023 %}).

Reviewer: ```suggestion
With the advent of Next-Generation Sequencing technologies around 2010, sequencing became significantly faster and more affordable. Today, the reduced sequencing costs, combined with increased computational resources compared to the early 2000s, make it possible to properly realize true pangenomes. This technological leap led to the release of the first human pangenome draft in 2023 ({% cite Liao2023 %}).
```

>
> 1. TOC
> {:toc}
>
{: .agenda}

# Data upload

To identify core and accessory genes across multiple bacterial strains, you will need the genome sequences in FASTA format of each strain.

> <hands-on-title>Data upload</hands-on-title>
>
> 1. Create a new history for this tutorial. Give it a name like `Pangenome Analysis`.
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
> 2. {% tool [Import](upload1) %} the following files from [Zenodo](https://doi.org/10.5281/zenodo.21339040).
>
>    ```
>    https://zenodo.org/records/21339041/files/10403S.fasta
>    https://zenodo.org/records/21339041/files/AUF.fasta
>    https://zenodo.org/records/21339041/files/EGD-e.fna
>    https://zenodo.org/records/21339041/files/EGD.fasta
>    https://zenodo.org/records/21339041/files/F6900.fna
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
> 3. Create a dataset collection containing the uploaded FASTA files. Give it a name like `Assemblies`.
>
>    {% snippet faqs/galaxy/collections_build_list.md %}
>
{: .hands_on}

# Annotate the Strains with Prokka

> You will use Prokka to annotate all bacterial strains simultaneously using the dataset collection. Prokka will scan the DNA sequences to locate genomic features, such as genes, and assign them their names and biological functions.

Reviewer: ```suggestion
You will use Prokka to annotate all bacterial strains simultaneously using the dataset collection. Prokka analyzes the DNA sequences to identify genomic features, such as genes, and assigns each a name and predicted biological function.
```

> <hands-on-title>Annotate Strains</hands-on-title>
>
> 1. Select `Tools` in the left sidebar and search for {% tool [Prokka](toolshed.g2.bx.psu.edu/repos/crs4/prokka/prokka/1.14.6+galaxy2) %} in the list that appears. Select it to open the tool. 
>
> 2. Within Prokka, select the following parameters (leave everything else unchanged):
>    - {% icon param-file %} *"Contigs to annotate"*: Click on the *"Dataset collection"* icon and select your `Assemblies` dataset collection
>    - *"Genus name"*: Enter `Listeria`
>    - *"Species name"*: Enter `monocytogenes` 
>    - *"Additional outputs"*: Deselect all options except `Annotation in GFF3 format, containing both sequences and annotations (.gff)`
>   
> 3. Run the tool. 
>
{: .hands_on}

## Examine the Output

Once Prokka has finished, two new dataset collections will appear in your history:

> - One containing the GFF3 output files for each strain. This is the dataset collection you will use further on. Take a look at each output file in this collection.

Reviewer: ```suggestion
- One containing the GFF3 output files for each strain. This is the dataset collection you will use further on. Examine each output file in this collection.
```

- One containing the logs of each run.

> <details-title>What is a GFF3 file?</details-title>
>
> **GFF3** stands for **General Feature Format (version 3)**. It is a plain text file used in bioinformatics to describe the exact locations of genomic features. The file contains 9 tab-separated columns that provide information for each feature ([official specification](http://www.sequenceontology.org/gff3.shtml)).
>
> Standard GFF3 files only contain the coordinates of genomic features. However, Prokka attaches the raw DNA sequence in FASTA format at the bottom of the GFF3 file (starting with a ##FASTA line in the file). This is exactly what the next tool requires to build the pangenome.
>
{: .details}

# Calculate the Pangenome with Roary

> Now you will run Roary to compare the genes identified across all strains by Prokka. Roary will cluster the genes based on their sequence similarity to construct the pangenome by identifying the core genes and the accessory genes.

Reviewer: ```suggestion
Now you will run Roary to compare the genes identified across all strains by Prokka. Roary clusters the genes according to sequence similarity, thereby identifying the core and accessory genes that constitute the pangenome.
```

> <hands-on-title>Identify Core and Accessory Genes</hands-on-title>
>
> 1. Select `Tools` in the left sidebar and search for {% tool [Roary](toolshed.g2.bx.psu.edu/repos/iuc/roary/roary/3.13.0+galaxy3) %} in the list that appears. Select it to open the tool. 
>
> 2. Within Roary, select the following parameters (leave everything else unchanged):
>    - *"Individual gff files or a dataset collection"*: Select `Collection` in the dropdown
>      - {% icon param-file %} *"Dataset collection to submit to Roary"*: Select the collection containing the GFF3 output files from the previous step.
>    - *"Additional outputs"*: Tick `Accessory binary genes in newick format`
>   
> 3. Run the tool. 
>
{: .hands_on}

## Examine the Output

> Once Roary has finished, four different output files will appear in your history. Take a look at each output file:

Reviewer: ```suggestion
Once Roary has finished, four different output files will appear in your history. Examine each output file:
```

- {% icon param-file %} **BIONJ Tree** file: This is an initial draft tree.
- {% icon param-file %} **MaxLikelihood Tree** file: Your main result. It contains the final phylogenetic tree in Newick format.
- {% icon param-file %} **MaxLikelihood Distance Matrix** file: Shows the calculated evolutionary distance between every possible pair of your strains.
- {% icon param-file %} **Occurrence Frequencies in Bootstrap Trees** file: Contains the statistical support for the tree's branches.
- {% icon param-file %} **Report and Final Tree** file: A summary of the entire run and a text-based visual of the tree.

## Visualize the Phylogenetic Tree

Now that you have a phylogenetic tree from your bacterial strains, you can view it visually. 

> <hands-on-title>Visualize the Phylogenetic Tree</hands-on-title>
>
> 1. Select the {% icon param-file %} **MaxLikelihood Tree** file in your history and click the {% icon galaxy-eye %} **View** icon.
>
> 2. By default, the file is displayed in {% icon galaxy-eye %} **Preview** mode. In the top tab bar, select {% icon galaxy-barchart %} **Visualize**. 
>
> 3. Select the `Phylogenetic Tree Browser`
>
{: .hands_on}

> You should see a phylogenetic tree like in Figure 3. Zoom in to look at each branch.

Reviewer: ```suggestion
The resulting phylogenetic tree should resemble Figure 3. Zoom in to examine individual branches.
```

<figure style="text-align: center;">
  <img src="./images/phylo_tree.png" alt="Phylogenetic tree of the five analyzed bacterial strains based on core genome" style="width:45%;">
  <figcaption>
    <strong>Figure 3: Phylogenetic tree of the five analyzed bacterial strains based on the core genome.</strong> Visualization of the evolutionary relationship and genetic divergence among the selected strains. The topology reveals two distinct groupings: (A) Strains 10403S, AUF, and EGD cluster closely together (their short branch lengths confirm that they share a near-identical core genome), and (B) Strains F6900 and EGD-e, which branch off into a separate cluster. This demonstrates a higher genetic distance and evolutionary divergence, both from the first group and from one another.
  </figcaption>
</figure>

> <comment-title>What can we deduce from these results?</comment-title>
>

> > - Strains that appear highly similar in this tree might have completely different traits in reality. This is because variations can exist within their accessory genes.

Reviewer: ```suggestion
> - Strains that appear highly similar in this tree may nonetheless differ substantially in phenotype. This is because variations can exist within their accessory genes.
```
